backend/services/inventory/api/http/routing.go

```diff
@@ -39,6 +39,15 @@ func AllowHeaderOptionsGenerator(methods []string) gin.HandlerFunc {
 	}
 }
 
+func rewritePathPrefix(old, new string) gin.HandlerFunc {
+	n := len(old)
+	return func(ctx *gin.Context) {
+		if strings.HasPrefix(ctx.Request.URL.Path, old) {
+			ctx.Request.URL.Path = new + ctx.Request.URL.Path[n:]
+		}
```

Comment:
I wonder what would be faster, Replace with limit 1? have you given strings.CutPrefix a look?

Reply:
I was actually looking into the implementation of the strings.Replace, but decided to go with this approach to save scan for counting occurrences and make it position dependent while only using a single allocation.